tasks/pull-disk-images/pull-disk-images.yaml

```diff
@@ -0,0 +1,88 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: pull-disk-images
+  labels:
+    app.kubernetes.io/version: "0.1.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task to pull disk images from quay using oras pull
+  params:
+    - name: snapshotPath
+      type: string
+      description: Path to the JSON string of the Snapshot spec in the data workspace
+    - name: concurrentLimit
+      type: string
+      description: The maximum number of images to be pulled at once
+      default: 3
+  workspaces:
+    - name: data
+      description: |
+        The workspace where the disk images will be written to. This workspace should be large (on the order
+        of 100G).
+  steps:
+    - name: pull-images
+      image:
+        quay.io/konflux-ci/release-service-utils:26f77b0933b19e7318bacf5dee297a6c113bb57e
+      script: |
+        #!/usr/bin/env bash
+        set -eux
+
+        SNAPSHOT_FILE="$(workspaces.data.path)/$(params.snapshotPath)"
+        if [ ! -f "${SNAPSHOT_FILE}" ] ; then
+            echo "No valid snapshot file was provided."
+            exit 1
+        fi
+
+        # Declare a location in the same subdirectory as the pipeline CRs to save the images to
+        DISK_IMAGE_DIR="$(dirname ${SNAPSHOT_FILE})"
+
+        process_component() { # Expected argument is [component json]
+            COMPONENT=$1
+            PULLSPEC=$(jq -r '.repository' <<< "${COMPONENT}")
+            DESTINATION="${DISK_IMAGE_DIR}/$(jq -r '.staged.destination' <<< "${COMPONENT}")/FILES"
+            mkdir -p "${DESTINATION}"
+            DOWNLOAD_DIR=$(mktemp -d)
```

Comment:
I had some issues in other tasks using `mktemp` without passing to it a writable directory due to default Pod permissions in the prod cluster. Maybe set a tmp dir in the `data` workspace?

Reply:
hmm, but `mktemp` puts it in `/tmp`. From the task logs
```
DOWNLOAD_DIR=/tmp/tmp.9p7xuOjK4T
```
We already write to `/tmp` in apply-mapping, create-internal-request, create-pyxis-image, publish-pyxis-repository, send-slack-notification, populate-release-notes-images, and push-rpm-manifests-to-pyxis, so I think it will be okay?

Reply:
Indeed. If it already works in other tasks, no concerns anymore.

Reply:
I think the issue was actually the mktemp was not used and instead we were writing to /etc, here:

Here: https://github.com/hacbs-release/app-interface-deployments/pull/110/files

Or did you mean something else? Either way, AFAIK, mktemp should be the safe thing to do.